README.md

```diff
@@ -53,72 +66,66 @@ inet:port_number()}`, are used to discover the rest of the cluster.
+close(ered_ref()) -> ok.
 ```
 
 Stop the main process. This will also stop the cluster handling
 process and in turn disconnect and stop all clients.
 
-### `command/3,4`
+### `command/2,3,4`
 
 ```Erlang
-command(server_ref(), command(), key()) -> reply().
-command(server_ref(), command(), key(), timeout()) -> reply().
+command(ered_ref(), command()) -> reply().
+command(ered_ref(), command(), timeout()) -> reply().
+command(ered_ref(), command(), key()) -> reply().
+command(ered_ref(), command(), key(), timeout()) -> reply().
 ```
 
-Send a command to the cluster. The command will be routed to
-the correct node client based on the provided key.
+Send a command. In cluster mode, the command will be routed to
+the correct node based on the provided key.
 If the command is a single command then it is represented as a
-list of binaries where the first binary is the command
+list of binaries where the first binary is the command name
 to execute and the rest of the binaries are the arguments.
 If the command is a pipeline, e.g. multiple commands to executed
 then they need to all map to the same slot for things to
 work as expected.
-
-`command/3` is the same as setting the timeout to infinity.
+For cluster clients, a key must be provided.
+Omitting timeout is the same as setting the timeout to infinity.
 
 ### `command_async/4`
 
 ```Erlang
-command_async(server_ref(), command(), key(), fun((reply()) -> any())) -> ok.
+command_async(ered_ref(), command(), fun((reply()) -> any())) -> ok.
+command_async(ered_ref(), command(), key(), fun((reply()) -> any())) -> ok.
 ```
 
-Like command/3,4 but asynchronous. Instead of returning the reply, the reply
-function is applied to the reply when it is available. The reply function
-runs in an unspecified process.
+Like command/2,3,4 but asynchronous. Instead of returning the reply, the reply
+function is applied to the reply when it is available. The reply function runs
+in an unspecified process and should not hang or perform any lengthy task.
 
 ### `command_all/2,3`
 
 ```Erlang
-command_all(server_ref(), command()) -> [reply()].
-command_all(server_ref(), command(), timeout()) -> [reply()].
+command_all(cluster_ref(), command()) -> [reply()].
+command_all(cluster_ref(), command(), timeout()) -> [reply()].
 ```
 
 Send the same command to all connected primary nodes.
 
-### `command_client/2,3`
-
-```Erlang
-command_client(client_ref(), command()) -> reply().
-command_client(client_ref(), command(), timeout()) -> reply().
-```
-
-Send the command to a specific client without any client routing.
-
-### `command_client_async/3`
-
-```Erlang
-command_client_async(client_ref(), command(), reply_fun()) -> ok.
-```
-
-Send command to a specific client in asynchronous fashion. The
-provided callback function will be called with the reply. Note that
-the callback function will executing in the client process and
-should not hang or perform any lengthy task.
-
 ### `get_clients/1`
 
 ```Erlang
```

Comment:
I believe we should replace `server_ref()` with `cluster_ref()` in the API descriptions below too, for both `get_clients/1` and `get_addr_to_client_map/1`.

Reply:
Good point. No i don't know aboit a trick.